src/native/libs/System.Security.Cryptography.Native/pal_evp_kdf.c

```diff
@@ -201,18 +201,26 @@ static int32_t HkdfCore(
 
         size_t keyLengthT = Int32ToSizeT(keyLength);
         size_t destinationLengthT = Int32ToSizeT(destinationLength);
-        size_t saltLengthT = Int32ToSizeT(saltLength);
-        size_t infoLengthT = Int32ToSizeT(infoLength);
 
-        OSSL_PARAM params[] =
+        OSSL_PARAM params[6] = {{0}};
+        int i = 0;
+        params[i++] = OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_KEY, (void*)key, keyLengthT);
+        params[i++] = OSSL_PARAM_construct_utf8_string(OSSL_KDF_PARAM_DIGEST, algorithm, 0);
+
+        if (salt != NULL && saltLength > 0)
         {
-            OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_KEY, (void*)key, keyLengthT),
-            OSSL_PARAM_construct_utf8_string(OSSL_KDF_PARAM_DIGEST, algorithm, 0),
-            OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_SALT, (void*)salt, saltLengthT),
-            OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_INFO, (void*)info, infoLengthT),
-            OSSL_PARAM_construct_int(OSSL_KDF_PARAM_MODE, &operation),
-            OSSL_PARAM_construct_end(),
-        };
+            size_t saltLengthT = Int32ToSizeT(saltLength);
+            params[i++] = OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_SALT, (void*)salt, saltLengthT);
+        }
+
+        if (info != NULL && infoLength > 0)
+        {
+            size_t infoLengthT = Int32ToSizeT(infoLength);
+            params[i++] = OSSL_PARAM_construct_octet_string(OSSL_KDF_PARAM_INFO, (void*)info, infoLengthT);
+        }
+
+        params[i++] = OSSL_PARAM_construct_int(OSSL_KDF_PARAM_MODE, &operation);
+        params[i++] = OSSL_PARAM_construct_end();
 
         if (EVP_KDF_derive(ctx, destination, destinationLengthT, params) <= 0)
```

Comment:
`assert(i < 6)`, though possibly introducing a local const to control the alloc/verify?

Reply:
There is no such thing as a "local const" in C, it still counts as a VLA which I wanted to avoid since it makes things weird. See https://stackoverflow.com/a/62035923/492405 for more info.

We can use a `#define` though.

Reply:
Okay, but that made the build cranky when the `#define` isn't used because we build against non-portable OpenSSL 1.1. So, "type the number out" it is.